### PR TITLE
[backend] rework document graphql interface/types #1235

### PR DESCRIPTION
**Pull request by jpkha**
# Context
> Briefly describe the purpose and context of this PR. Include relevant screenshots or screen recordings for the product team.

Rework Document to avoid typename conflict

## Related issues

- Related to #1235 

# Checklist

## Quality

- [x] I consider this work complete and ready for review
- [x] I tested all features and edge cases
- [x] I refactored code where necessary to improve quality
- [x] I made sure my code meets development guidelines
- [x] I performed a self-review of my code

## Testing

- [ ] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [ ] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [ ] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [ ] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

File: apps/portal-api/src/modules/services/document/document.resolver.ts [hunk 1]
```diff
@@ -154,6 +159,30 @@ const resolvers: Resolvers = {
...
+        logApp.warn(
+          `Document resolver type - Unresolved integration type ${integrationType}`
+        );
+        return INTEGRATION_MAPPINGS[integrationType] ?? 'DefaultDocument';
+      }
+      logApp.warn(
+        `Document resolver type - Unresolved document type ${document.type}`
+      );
+      return 'DefaultDocument';
+    },
```
Comment (Kroustille) [inline apps/portal-api/src/modules/services/document/document.resolver.ts]:
```suggestion
      } else if (document.type === OPENCTI_INTEGRATION_FEED_DOCUMENT_TYPE) {
        const integrationType = await loadIntegrationType(document.id);
        const responseType = INTEGRATION_MAPPINGS[integrationType]
        if (responseType) {
          return responseType;
        }
      }
      logApp.warn(
        `Document resolver type - Unresolved document type ${document.type}`
      );
      return 'DefaultDocument';
    },
```
